python/shotgun_model/shotgun_query_model.py

```diff
@@ -0,0 +1,979 @@
+        of some kind.
+
+    :signal data_refreshed(bool): Emitted whenever the model has been updated
+        with fresh shotgun data. The boolean indicates that a change in the
+        model data has taken place as part of this process. If the refresh fails
+        for some reason, this signal may not be emitted.
+
+    :signal data_refresh_fail(str): Emitted in the case the refresh fails for
+        some reason, typically due to the absence of an internet connection.
+        This signal could for example be used to drive a "retry" button of some
+        kind. The str parameter carries an error message with details about why
+        the refresh wasn't successful.
+
+    Constants
+    ---------
+
+    :constant SG_DATA_ROLE: Custom model role that holds the shotgun data
+        payload
+
+    :constant IS_SG_MODEL_ROLE: Used to identify model items related to Shotgun
+        data
+
+    Class Members
+    -------------
+
+    Subclasses can set a ``SG_QUERY_MODEL_ITEM_CLASS`` class member that
+    identifies the class of item to use when constructing the model. This is
+    also used during deserialization to create model items. The default is
+    the ``ShotgunStandardItem``. If overriding, the class must subclass from
+    ``ShotgunStandardItem``.
+
+        SG_QUERY_MODEL_ITEM_CLASS = ShotgunStandardItem
+
+    Subclasses must also define the ``SG_DATA_UNIQUE_ID_FIELD`` class member.
+    This is used to specify the field in the Shotgun payload that is used to
+    uniquely identify an item in the model.
+
+        SG_DATA_UNIQUE_ID_FIELD = "id"
+
+    """
+
+    # ---- signals
+
+    # signal emitted after the model's sg query is changed
+    query_changed = QtCore.Signal()
+
+    # signal emitted after the model loads cache data
+    cache_loaded = QtCore.Signal()
+
+    # signal emitted before the model starts to refresh its shotgun data
+    data_refreshing = QtCore.Signal()
+
+    # signal emitted after the model is updated with fresh shotgun data
+    data_refreshed = QtCore.Signal(bool)
+
+    # signal emitted in the case the refresh fails
+    data_refresh_fail = QtCore.Signal(str)
+
+    # ---- internal constants
+
+    # please do not access directly but instead use the helper
+    # methods provided! We may change these constants without prior notice.
+    SG_DATA_ROLE = QtCore.Qt.UserRole + 1
+    IS_SG_MODEL_ROLE = QtCore.Qt.UserRole + 2
+
+    # ---- "abstract" class members
+
+    # subclasses should define a unique id field from the SG data for use in
+    # caching and associating data with items in the model
+    SG_DATA_UNIQUE_ID_FIELD = None
+
+    # subclasses should define the class to use when loading items from disk
+    SG_QUERY_MODEL_ITEM_CLASS = ShotgunStandardItem
+
+    # ---- caching/serialization related costants
+
+    # magic number for IO streams
+    _FILE_MAGIC_NUMBER = 0xDEADBEEF
+
+    # version of binary format
+    _FILE_VERSION = 22
+
+    def __init__(self, parent, bg_task_manager=0):
```

Comment:
I think this is leftover, cut/paste error from `schema_generation=0` arg. Fixed and updated docstring.